src/transformers/models/xlm_roberta_xl/modeling_xlm_roberta_xl.py

```diff
@@ -1119,6 +1119,12 @@ def prepare_inputs_for_generation(self, input_ids, past_key_values=None, attenti
         if attention_mask is None:
             attention_mask = input_ids.new_ones(input_shape)
 
+        # Create missing `position_ids` on the fly
+        position_ids = None
+        if model_kwargs.get("position_ids") is None:
+            position_ids = create_position_ids_from_input_ids(input_ids, padding_idx=self.config.pad_token_id)
+            position_ids = position_ids  # placed in kwargs for further processing (see below)
```

Comment:
ultra nit: this line is redundant a bit

Reply:
wow, i don't know why I added this 😭 thank you!

Reply:
BTW, some models' position ids are different (`Roberta` one is an example). It should use `create_position_ids_from_input_ids`, but `Roberta`'s `prepare_inputs_for_generation` doesn't use it and use the common one. 

This is another story - but I kind think it's a bug ...

Reply:
yeah, that is the reason we need one method to `create_position_ids` which will be used everywhere. But last time it resulted in so many failures, especially when checking TF/JAX equivalence. Also might be quite breaking change :(

Oke, we'll take it noted for now